### DISCUSSION TITLE
Is there an Array string in Payload CMS?

### DISCUSSION THREAD

**Discussion by IRediTOTO** [payloadcms/payload#1791] [Q&A] [answered]
Hi, I tried to read the documentation, but I couldn't find it. Is there something to save array string to database? 
I just want save array of strings
Ex: `["a","b","c"]`
I don't want save array of objects like this, just need array of strings here
![image](https://user-images.githubusercontent.com/15695252/210272435-2b0022c9-de6e-40ba-99a9-8430686ec1b3.png)

**Comment by JessRynkar** [comment]
Hi @IRediTOTO, this isn't currently supported but would definitely be useful in many cases. This question was also discussed on discord [here](https://discord.com/channels/967097582721572934/1050348908729073685/1050348908729073685). A potential workaround solution would be to:
1. Create a hidden `preferences` field
2. On save, map the original array and return as an array of strings 
3. Pass this new array to your hidden field

There might be more appropriate workarounds too if you want to share more information on what you're looking to achieve, let us know if you need further help here!

**Reply by IRediTOTO** [reply]
Thank you for replied me and let me know. Ok fine, but I will wait for a that feature like that. This pretty basic, I think any Payload newbie will ask himself: where is the array??? :D

**Reply by gaboluque** [reply]
Payload newbie here, I think this could be really useful.

**Reply by jacobsfletch** [reply]
The `json` field might work for you: https://payloadcms.com/docs/fields/json#json-field

**Comment by jacobsfletch** [comment, marked as answer]
The`json` field might work for you: https://payloadcms.com/docs/fields/json#json-field

**Comment by akshayxemo** [comment]
Hey @IRediTOTO I am also a payload newbie and I have came across the same issue and after so many tries and seeing this discussion I have found this way to be able to store the data.  I know its already been answered but I still want to share this one hope this will be helpful or correct me if I am following the wrong approach.

1. if you want to store multiple same string in an array the following one will work for you. 
2. If you want all the strings are in the array will be unique then simply use type:"text" with hasMany: true, it will work.

```TypeScript
import { CollectionConfig } from "payload/types";

const Users: CollectionConfigs = {
  slug: "users",
  fields: [
     {
      name: "restrictions",
      label: "Restrictions",
      type: "array",
      fields: [
        {
          name: "preferences",
          type: "text",
          required: true,
        },
      ],
      hooks: {
        afterRead: [
          ({ value, originalDoc }) => {
            value = originalDoc?.preferences?.map((value: string) => {
              return { preferences: value };
            });
            return value;
          },
        ],
      },
    },
    {
      name: "preferences",
      type: "text",
      hasMany: true,
      admin: {
        readOnly: true,
      },
      hooks: {
        beforeChange: [
          ({ value, siblingData}) => {
            const preferences =
              siblingData.restrictions.map((el: any) => el.preferences) || [];
            delete siblingData["restrictions"];
            value = preferences;
            return value;
          },
        ],
      },
    },
  ]
}
```

**Comment by willemmulder** [comment]
In 2025, there is now the 'hasMany' option for a.o. the text, select and number field, which should do exactly what you want 😃  See e.g. https://payloadcms.com/docs/fields/text @IRediTOTO

**Reply by ziyafenn** [reply]
This is great, but why on earth does it create a separate table to store array items instead of creating []text array type column

**Reply by willemmulder** [reply]
I don't know - in Mongo it doesn't AFAIK. Do you use Postgres?

**Reply by ziyafenn** [reply]
yeah i use postgres, i ended up extending table with my own postgres column outside of the payload
